airflow/utils/net.py

```diff
@@ -0,0 +1,16 @@
+from airflow.configuration import conf, AirflowConfigException
+import socket
+
+
+def get_hostname(default=socket.getfqdn):
+    """
+    A replacement for `socket.getfqdn` that allows configuration to override it's value.
+
+    :param callable|str default: Default if config does not specify. If a callable is given it will be called.
+    """
+    try:
+        # If this is called `get`, why does it not match python semantics?
+        hostname = conf.get('core', 'hostname')
+    except AirflowConfigException:
+        hostname = callable(default) and default() or default
```

Comment:
Let's not chain too many of these. It's a bit harder to comprehend

Reply:
Also, you can do `conf.get('core', 'hostname', fallback=callable(default) and default() or default)` and not need the try/except at all.

Reply:
@ashb Nice. Will update that.

Reply:
(tomorrow)

Reply:
updated (late)

Reply:
@ashb I tried that, still raises, don't really have time to look into it too far tbh, wish I did though

Reply:
Is there a reason this isn't a simple dictionary interface by the way?